genie-agent/src/main/java/com/netflix/genie/agent/execution/statemachine/actions/SetUpJobAction.java

```diff
@@ -119,40 +125,57 @@ protected Events executeStateAction(
 
     @Override
     protected void executeStateActionCleanup(final ExecutionContext executionContext) {
+
+        final JobFolderCleanupOption cleanupOption = this.agentOptions.getJobFolderCleanUpOption();
+
         final File jobDirectory = executionContext.getJobDirectory();
         final Path jobDirectoryPath = jobDirectory.toPath();
 
-        // TODO: this is a simple and safe cleanup strategy, 3 patterns are added that match any files in the
-        // dependencies directory of any entity by absolute path.
-        // A different implementation would allow override via server or CLI options, or collect the files list as
-        // they are downloaded and compile an exact list using Pattern.quote()
-        final RegexRuleSet cleanupWhitelist = RegexRuleSet.buildWhitelist(
-            (Pattern[]) Lists.newArrayList(
-                PathUtils.jobClusterDirectoryPath(jobDirectory, ".*"),
-                PathUtils.jobCommandDirectoryPath(jobDirectory, ".*"),
-                PathUtils.jobApplicationDirectoryPath(jobDirectory, ".*")
-            )
-                .stream()
-                .map(PathUtils::jobEntityDependenciesPath)
-                .map(Path::toString)
-                .map(pathString -> pathString + "/.*")
-                .map(Pattern::compile)
-                .toArray(Pattern[]::new)
-        );
+        //No clean up needed
+        if (cleanupOption == JobFolderCleanupOption.NO_CLEANUP) {
+            log.info("Job folder not deleted.");
+        } else if (cleanupOption == JobFolderCleanupOption.DELETE_JOB_FOLDER) { //Delete the job folder
+            try {
+                log.debug("Deleting {}", jobDirectory);
+                FileSystemUtils.deleteRecursively(jobDirectoryPath);
+                log.info("Deleted job folder.");
+            } catch (final IOException e) {
+                log.warn("Failed to delete job folder: {}", jobDirectoryPath.toAbsolutePath().toString(), e);
+            }
+        } else { //Remove only the dependencies
+
+            // TODO: this is a simple and safe cleanup strategy, 3 patterns are added that match any files in the
```

Comment:
You have this as a TODO. Do you intend to change it or leave it as is. If you're leaving it then I'd remove the TODO